paddlenlp/transformers/model_utils.py

```diff
@@ -1530,6 +1519,22 @@ def from_pretrained_v2(
         if not os.path.exists(os.path.join(cache_dir, CONFIG_NAME)):
             config.save_pretrained(cache_dir)
 
+        # PretrainedConfig auto contains dtype field
```

Comment:
https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py#L2562

Reply:
paddlenlp 中的参数初始化方式没有完全照搬 transformers ：直接在派生类模型当中编写`init_weights`方法，从而没办法在基类中控制是否`_init_weights`：

* pytorch 中init_weights 的方式

```python
    def init_weights(self):
        """
        If needed prunes and maybe initializes weights. If using a custom `PreTrainedModel`, you need to implement any
        initialization logic in `_init_weights`.
        """
        # Prune heads if needed
        if self.config.pruned_heads:
            self.prune_heads(self.config.pruned_heads)

        if _init_weights:
            # Initialize weights
            self.apply(self._initialize_weights)

            # Tie weights should be skipped when not initializing all weights
            # since from_pretrained(...) calls tie weights anyways
            self.tie_weights()
```

模型中的参数是在`._initialize_weights`中处理（有做二次调用）。

* paddle 中的参数初始化

```python
    def __init__(self, config: BertConfig):
        super(BertModel, self).__init__(config)
        self.apply(self.init_weights)

    def init_weights(self, layer):
        """Initialization hook"""
        if isinstance(layer, (nn.Linear, nn.Embedding)):
            # In the dygraph mode, use the `set_value` to reset the parameter directly,
            # and reset the `state_dict` to update parameter in static mode.
            if isinstance(layer.weight, paddle.Tensor):
                layer.weight.set_value(
                    paddle.tensor.normal(
                        mean=0.0,
                        std=self.config.initializer_range,
                        shape=layer.weight.shape,
                    )
                )

        elif isinstance(layer, nn.LayerNorm):
            layer._epsilon = self.config.layer_norm_eps
```

如果想要和 transformers 保持完全一致的话，那就需要在每个模型中添加 `_init_weights`的条件判断，这样对模型的侵入性太大。

其实，paddle 中的self.apply 就相当于pytorch中的`init_weights`，故可以调整self.apply来实现此功能。

Reply:
经过和 @ZHUI 讨论，为了和HF 保持一致，后续会陆续将调整所有模型的数据初始化方式，在此次 PR 当中只针对于 Bloom、GPT、LLaMA、GLM 四个模型进行调整。